src/reactiveData.ml

```diff
@@ -206,7 +206,60 @@ module DataList = struct
       end;
       Array.to_list a
 
-  let merge p l = List.fold_left (fun l x -> merge_p x l) l p
+  (* accumulates into acc i unmodified elements from l *)
+  let rec linear_merge_fwd i l ~acc =
+    assert (i >= 0);
```

Comment:
You already check that in the checking function, and this function is not exposed, so you can remove it. :)